auth/user-authentication-middleware.go

```diff
@@ -0,0 +1,45 @@
+// Author: Maximilian Floto
+package auth
+
+import (
+	"net/http"
+)
+
+// Takes as parameters the function serving the endpoint, the minimum role, an array of functions that are not subject to authentication
+func UserAuthMiddleware(f func(http.ResponseWriter, *http.Request), role Role, unrestrictedMethods []string) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

Comment:
Is it just GH, or why is there so much indentation?

Reply:
For me it's just one tab / four spaces